src/Storages/VirtualColumnUtils.h

```diff
@@ -74,12 +79,11 @@ struct VirtualsForFileLikeStorage
     std::optional<size_t> size { std::nullopt };
     const String * filename { nullptr };
     std::optional<Poco::Timestamp> last_modified { std::nullopt };
-
 };
 
 void addRequestedFileLikeStorageVirtualsToChunk(
     Chunk & chunk, const NamesAndTypesList & requested_virtual_columns,
-    VirtualsForFileLikeStorage virtual_values);
+    VirtualsForFileLikeStorage virtual_values, ColumnsDescription columns = {}, ContextPtr context = {});
```

Comment:
It's not good to set a default value for `context`, because default value is `nullptr` and inside `addRequestedFileLikeStorageVirtualsToChunk` we always call `context->getSettingsRef()` without checking if it's `nullptr`.

Reply:
And also `ColumnsDescription columns = {}` -> `const ColumnsDescription & column = {}`

Reply:
I would remove default values for both `context` and `columns`. Tis function is usedonly in file-like storages and we should always provide these arguments. For `ObjectStorageQueueSource` it can also make sense to support hive-partitioning in the future (no need for now)

Reply:
I guess integration tests with `s3Queue` are failimig because of segfault due to `context = nullptr` here